spec/controllers/election/candidates_controller_spec.rb

```diff
@@ -0,0 +1,163 @@
+require 'rails_helper'
+
+RSpec.describe Election::CandidatesController, type: :controller do
+  # Override get_commit
+  before(:each) do
+    allow_any_instance_of(ApplicationController).to receive(:get_commit).and_return(true)
+  end
+
+  # Build an election
+  let(:election) { create(:election) }
+  before(:each) do
+    allow(Election).to receive(:current).and_return(election)
+  end
+
+  # Testing objects
+  let(:user) { create(:user) }
+  let(:search_post) { create(:post) }
+  let(:not_owner) { create(:user) }
+  let(:candidate) do
+    create(:candidate, profile: user.profile, election: election, post: search_post)
+  end
+
+  context 'when not signed in' do
+    describe 'GET #index' do
+      it 'redirects' do
+        get :index
+
+        expect(response).to be_redirect
+      end
+    end
+  end
+
+  context 'when signed in as user' do
+    before do
+      sign_in user
+      search_post
+    end
+
+    it "should have a current_user" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.